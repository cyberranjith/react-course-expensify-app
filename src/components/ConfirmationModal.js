import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
    <Modal
        isOpen={props.showModal}
        onRequestClose={props.closeConfirmation}
        contentLabel={props.confirmationText}
        closeTimeoutMS={200}
        ariaHideApp={false}
        className="modal"
    >
        <h3 className="modal__title">{props.confirmationText}</h3>
        <button onClick={props.performConfirmedAction} className="modal__button">Yes</button>
        <button onClick={props.closeConfirmation} className="modal__button">No</button>
    </Modal>
);

export default ConfirmationModal;