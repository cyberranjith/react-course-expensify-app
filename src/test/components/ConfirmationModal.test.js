import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

let wrapper, performConfirmedAction, closeConfirmation;

beforeEach ( () => {
    performConfirmedAction = jest.fn();
    closeConfirmation = jest.fn();
    wrapper = shallow(
        <ConfirmationModal
            showModal={true}
            confirmationText="Are you sure want to remove this expense?"
            performConfirmedAction={performConfirmedAction}
            closeConfirmation={closeConfirmation}
        />
    );
});

test('should render Confirmation Modal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should perform the confirmed action on clicking Yes', () => {
    // Click the Yes button
    wrapper.find('button').at(0).simulate('click');
    expect(performConfirmedAction).toBeCalled();
});

test('should call close confirmation on clicking No', () => {
    // Click the No button
    wrapper.find('button').at(1).simulate('click');
    expect(closeConfirmation).toBeCalled();
});