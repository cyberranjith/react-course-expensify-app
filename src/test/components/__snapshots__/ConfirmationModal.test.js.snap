// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Confirmation Modal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure want to remove this expense?"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Are you sure want to remove this expense?
  </h3>
  <button
    className="modal__button"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="modal__button"
    onClick={[Function]}
  >
    No
  </button>
</Modal>
`;
